RAGchain/benchmark/dataset/mr_tydi.py

```diff
@@ -0,0 +1,152 @@
+        """
+
+        # If ingest size too bit, It takes a long time.(Because corpus size is 1496126!)
+        # So we shuffle corpus and slice by ingest size for test.
+        # Put retrieval gt corpus in passages because retrieval retrieves ground truth in db.
+
+        make_retrieval_gt_passage = deepcopy(self.qa_data['positive_passages'])
+        make_passages = deepcopy(self.corpus)
+
+        if ingest_size is not None:
+            # ingest size must be larger than evaluate size.
+            if ingest_size >= self.eval_size:
+                make_passages = sklearn.utils.shuffle(make_passages)[:ingest_size]
+            else:
+                raise ValueError("ingest size must be same or larger than evaluate size")
+
+        # Remove duplicated passages between corpus and retrieval gt for ingesting passages faster.
+        make_retrieval_gt_passage = make_retrieval_gt_passage.apply(self.__make_corpus_passages)
+        make_retrieval_gt_passage, gt_ids = zip(*make_retrieval_gt_passage)
+        id_for_remove_duplicated_corpus = list(itertools.chain.from_iterable(gt_ids))
+
+        # Marking dupicated values in the corpus using retrieval_gt id.
+        mask = make_passages.isin(id_for_remove_duplicated_corpus)
+        # Remove duplicated passages
+        make_passages = make_passages[~mask.any(axis=1)]
+
+        make_retrieval_gt_passage = list(itertools.chain.from_iterable(make_retrieval_gt_passage))
+        passages = make_passages.apply(self.__make_corpus_passages, axis=1).tolist()
+        for gt_passage in make_retrieval_gt_passage:
+            passages.append(gt_passage)
+
+        for retrieval in retrievals:
+            retrieval.ingest(passages)
+        db.create_or_load()
+        db.save(passages)
+        # TODO: 다른 language로 ingest할때는 그전건 초기화 시켜야함.
+        # TODO: 공식문서 쓸때 내 velog link 걸기(영어로 번역)
+
+    def evaluate(self, **kwargs) -> EvaluateResult:
+        retrieval_gt = [[passage_id['docid'] for passage_id in passages] for passages in
+                        self.qa_data['positive_passages'].tolist()]
+
+        return self._calculate_metrics(
+            questions=self.qa_data['question'].tolist(),
+            pipeline=self.run_pipeline,
+            retrieval_gt=retrieval_gt,
+            **kwargs
+        )
+
+    def __make_corpus_passages(self, row):
+        # retrieval gt to passage for ingest.
+        if type(row) == list:
+            gt_passages = []
+            # gt_id for removing duplicated value
+            gt_id = []
+            for element in row:
+                gt_passages.append(Passage(
+                    id=element['docid'],
+                    content=element['text'],
+                    filepath=self.file_path,
+                    metadata_etc={'title': element['title']}
+                ))
+
+                gt_id.append(element['docid'])
+            return gt_passages, gt_id
```

Comment:
different return types in same function? 
Function must do one thing, one thing only.